subworkflows/local/profiling.nf

```diff
@@ -376,6 +380,31 @@ workflow PROFILING {
 
     }
 
+    if (params.run_kmcp) {
+
+            ch_input_for_kmcp = ch_input_for_profiling.kmcp
+                                .filter {
+                                    meta, reads, meta_db, db ->
+                                        if ( meta['instrument_platform'] == 'OXFORD_NANOPORE' ) log.warn "[nf-core/taxprofiler] Kmcp is only suitable for short-read metagenomic profiling, with much lower sensitivity on long-read datasets. Skipping kmcp for sample ${meta.id}."
+                                        meta_db['tool'] == 'kmcp' && meta['instrument_platform'] != 'OXFORD_NANOPORE'
+                                    }
+                                    .multiMap {
+                                    it ->
+                                        reads: [ it[0] + it[2], it[1] ]
+                                        db: it[3]
+
+                            }
+
+            KMCP_SEARCH (ch_input_for_kmcp.db, ch_input_for_kmcp.reads)
+            KMCP_PROFILE (KMCP_SEARCH.out.result,ch_input_for_kmcp.db, params.kmcp_mode)
```

Comment:
This won't work, if multiple KMCP databases are supplied, the wrong database will be linked with the wrong profiles between SEARCH and PROFILE (as they won't come out at the same time).

See what we have done for e.g. Kaiju or Centrifuge, when there is a two step process - i.e., the funky function thing

Reply:
Done